docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -0,0 +1,492 @@
+1. Create an intermediate CA for TLS server authentication
+1. Sign the intermediate CA using the root CA
+1. Configure policy for intermediate CA
+1. Generate and store leaf certificates for
+   Consul,
+   Vault,
+   PostgreSQL,
+   Kong (admin),
+   Kong (external),
+   Redis (v6),
+   Mosquitto
+1. Deploy the PKI to the respective services' secrets area
+1. Write the production Vault configuration (TLS-enabled)
+
+Consul bootstrapping:
+
+1. Write the production Consul configuration (TLS-enabled)
+   * HTTPS on port 8501
+   * Disable HTTP on 8500
+   * Disable DNS
+   * Specify ca_file, cert_file, and key_file
+1. Generate a Serf encryption key and save it to Vault
+1. Start consul and generate a management token
+1. Configure the Consul secrets engine in Vault
+
+Secrets seeding:
+
+1. Generate a PostgreSQL superuser password and save it to Vault
+1. Generate the Redis database password and save it to Vault
+1. Generate a Kong admin credential and save it to Vault
+
+Kong bootstrapping:
+
+1. Start up a temporary instance of Postgres and a temporary instance of Kong
+1. Set up routes to internal microservices
+1. Install externally facing TLS certificate
+1. Configure Kong API loopback protection on admin port
+
+Cleanup:
+
+1. Revoke temporary Vault root token
+1. Terminate all services that were previously started
+
+Done:
+1. Start a HTTP daemon to report HTTP "200 OK" status
+
+#### Vault
+
+Will wait for `install`
+and then start in production mode in a sealed state.
+
+#### Consul
+
+Will wait for `install`
+and then start in production mode.
+
+#### upgrade
+
+The upgrade semantics have not been defined
+as EdgeX currently does not have an LTS release.
+
+#### secretstore_ready
+
+Security-enabled:
+
+1. Unseal Vault
+1. Issue fresh access tokens for all services.
+1. Start its HEALTHCHECK and reports Vault's health as its own.
+
+No-security:
+
+1. Wait for Consul
+
+#### persistence_ready
+
+Security-enabled:
+
+1. Wait for Vault, Postgres, and Redis.
+
+No-security:
+
+1. Wait for Redis.
+
+#### ready_to_run
+
+1. Wait for `install`, `upgrade`, `secretstore_ready`, and `persistence_ready`.
+
+All other microservices will wait on `ready_to_run`.
+
+
+## Consequences
+
+### Benefits
+
+This ADR is expected to yield the following benefits after completion of the related engineering tasks:
+
+* Clear separation of the one-time initialization flows from the per-boot flows.
+
+* Reduced fragility in the framework startup flow.
+
+* Reduction in the variability of init-blocking methods currently employed
+  to block service startup until necessarily initialization has completed.
+
+* Deployment of a foundation required for secure EdgeX microservice communications.
+
+* Reduction in future work required to securely distribute EdgeX services across multiple nodes.
+
+### Breaking changes
+
+Instructions to reviewers: please point out the breaking changes.
```

Comment:
I assume the connection to consul from the services will be via HTTPS, correct? I think this is a breaking change since my Hanoi App or Device service is configured to use HTTP in the TOML (Registry) and command-line (Config Provider)